### PR TITLE
Fix stack

### PR DESCRIPTION
**Pull request by robcohen**
Stack was having issues importing from hedgehog, and it has been corrected.

### CODE REVIEW

**Comment by robcohen** [comment]
More strangeness. Hedgehog error returned, but went away only on a second run of `stack build`.

**Comment by robcohen** [comment]
This PR is desired after my neglecting to make proper edits on GH-6 to maintain stack build functionality alongside nix build.

File: stack-shell.nix [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
+  nixpkgs = builtins.fetchTarball {
+    url    = "https://github.com/NixOS/nixpkgs/archive/069bf7aee30faf7b3ed773cfae2154d761b2d6c2.tar.gz";
+    sha256 = "1c44vjb60fw2r8ck8yqwkj1w4288wixi59c6w1vazjixa79mvjvg";
+  };
+
+  pkgs = import nixpkgs { inherit config; };
+
+in 
 
 haskell.lib.buildStackProject {
```
Comment (ksaric) [inline stack-shell.nix]: Where does `haskell` come from?

Comment (robcohen) [reply stack-shell.nix]: I believe https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/haskell-modules

**Comment by robcohen** [comment]
```
stack build
Cloning d3a3dfc451725cc3e17f47677739ba3dc7d9bbee from https://github.com/input-output-hk/cardano-prelude
Cloning into '/home/coder/Nextcloud/IOHK/cardano-shell/.stack-work/downloaded/HLreYyMWoXp1'...
remote: Enumerating objects: 98, done.
remote: Counting objects: 100% (98/98), done.
remote: Compressing objects: 100% (60/60), done.
remote: Total 98 (delta 35), reused 86 (delta 28), pack-reused 0
Unpacking objects: 100% (98/98), done.
Cloning f66762e4c4fe484ed109d28ad541fc55cc841c2c from https://github.com/input-output-hk/canonical-json.git
Cloning into '/home/coder/Nextcloud/IOHK/cardano-shell/.stack-work/downloaded/Dk8pknEK6WOA'...
remote: Enumerating objects: 29, done.
remote: Total 29 (delta 0), reused 0 (delta 0), pack-reused 29
Unpacking objects: 100% (29/29), done.
Cloning e87fcd3fea18f664d6e4f39c11abb8cafbd25785 from https://github.com/input-output-hk/cardano-crypto
Cloning into '/home/coder/Nextcloud/IOHK/cardano-shell/.stack-work/downloaded/MZeO7rE4cJq7'...
remote: Enumerating objects: 1343, done.
remote: Total 1343 (delta 0), reused 0 (delta 0), pack-reused 1343
Receiving objects: 100% (1343/1343), 411.66 KiB | 4.29 MiB/s, done.
Resolving deltas: 100% (576/576), done.
Cloning 3d274c14ca3077c3a081ba7ad57c5182da65c8c1 from https://github.com/well-typed/cborg
Cloning into '/home/coder/Nextcloud/IOHK/cardano-shell/.stack-work/downloaded/hqLs3p1i6tie'...
remote: Enumerating objects: 139, done.
remote: Counting objects: 100% (139/139), done.
remote: Compressing objects: 100% (92/92), done.
remote: Total 4002 (delta 62), reused 100 (delta 46), pack-reused 3863
Receiving objects: 100% (4002/4002), 1.05 MiB | 10.67 MiB/s, done.
Resolving deltas: 100% (2072/2072), done.
Cloning 2c9e51804e8217dff89f5c32cbe0d79ce20bc508 from https://github.com/input-output-hk/haskell-hedgehog.git
Cloning into '/home/coder/Nextcloud/IOHK/cardano-shell/.stack-work/downloaded/27pmaoPEDNVl'...
remote: Enumerating objects: 24, done.
remote: Counting objects: 100% (24/24), done.
remote: Compressing objects: 100% (19/19), done.
remote: Total 1439 (delta 7), reused 17 (delta 5), pack-reused 1415
Receiving objects: 100% (1439/1439), 393.29 KiB | 4.74 MiB/s, done.
Resolving deltas: 100% (778/778), done.
Please ensure that commit 2c9e51804e8217dff89f5c32cbe0d79ce20bc508 exists within https://github.com/input-output-hk/haskell-hedgehog.git
Received ExitFailure 128 when running
Raw command: /usr/bin/git --git-dir=.git reset --hard 2c9e51804e8217dff89f5c32cbe0d79ce20bc508 --
Run from: /home/coder/Nextcloud/IOHK/cardano-shell/.stack-work/downloaded/27pmaoPEDNVl/
Standard error:

fatal: Could not parse object '2c9e51804e8217dff89f5c32cbe0d79ce20bc508'.

$ stack build
Cloning d094be301195fcd8ab864d793f114970426a4478 from https://github.com/avieth/plutus-prototype
Cloning into '/home/coder/Nextcloud/IOHK/cardano-shell/.stack-work/downloaded/KQSGhWYbOHSL'...
remote: Enumerating objects: 1156, done.
remote: Total 1156 (delta 0), reused 0 (delta 0), pack-reused 1156
Receiving objects: 100% (1156/1156), 2.88 MiB | 13.49 MiB/s, done.
Resolving deltas: 100% (746/746), done.
cardano-prelude-0.1.0.0: configure (lib)
cardano-prelude-0.1.0.0: build (lib)
cardano-prelude-0.1.0.0: copy/register
Building all executables for `cardano-shell' once. After a successful build of all of them, only specified executables will be rebuilt.
cardano-shell-0.1.0.0: configure (lib + exe)
Configuring cardano-shell-0.1.0.0...
cardano-shell-0.1.0.0: build (lib + exe)
Preprocessing library for cardano-shell-0.1.0.0..
Building library for cardano-shell-0.1.0.0..
[1 of 2] Compiling Lib              ( src/Lib.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Lib.o )
[2 of 2] Compiling Paths_cardano_shell ( .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/autogen/Paths_cardano_shell.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/Paths_cardano_shell.o )
Preprocessing executable 'cardano-shell-exe' for cardano-shell-0.1.0.0..
Building executable 'cardano-shell-exe' for cardano-shell-0.1.0.0..
[1 of 2] Compiling Main             ( app/Main.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/cardano-shell-exe/cardano-shell-exe-tmp/Main.o )
[2 of 2] Compiling Paths_cardano_shell ( .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/cardano-shell-exe/autogen/Paths_cardano_shell.hs, .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/cardano-shell-exe/cardano-shell-exe-tmp/Paths_cardano_shell.o )
Linking .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.2.0.1/build/cardano-shell-exe/cardano-shell-exe ...
cardano-shell-0.1.0.0: copy/register
Installing library in /home/coder/Nextcloud/IOHK/cardano-shell/.stack-work/install/x86_64-linux-tinfo6/custom-cardano-snapshot-N27vz4n30Tnn/8.4.4/lib/x86_64-linux-ghc-8.4.4/cardano-shell-0.1.0.0-Ibavo4TOHEy9zDpOGwSsvs
Installing executable cardano-shell-exe in /home/coder/Nextcloud/IOHK/cardano-shell/.stack-work/install/x86_64-linux-tinfo6/custom-cardano-snapshot-N27vz4n30Tnn/8.4.4/bin
Registering library for cardano-shell-0.1.0.0..
Completed 2 action(s).             
```

**Comment by ksaric** [comment]
@robcohen I still get the same error - `fatal: Could not parse object '2c9e51804e8217dff89f5c32cbe0d79ce20bc508'.`.
My comment https://github.com/input-output-hk/cardano-shell/pull/9#discussion_r228953655 is referring to something else that you need to figure out.

**Pull request merged**